core/arch/arm/kernel/thread.c

```diff
@@ -1479,11 +1493,11 @@ uint32_t thread_enter_user_mode(unsigned long a0, unsigned long a1,
 	uint32_t exceptions = 0;
 	uint32_t rc = 0;
 	struct thread_ctx_regs *regs = NULL;
-	unsigned long keys[2];
+	struct thread_pauth_keys *keys = NULL;
 
 	tee_ta_update_session_utime_resume();
 
-	tee_ta_get_pauth_key(keys);
+	keys = thread_get_pauth_keys();
```

Comment:
To get the pauth keys in the thread_enter_user_mode function, I have 2 choices ->
one which I have implemented here i.e reach user_mode_ctx from ts_get_current_session()
or 
-> uctx is known in all the places from where thread_enter_user_mode() is called from i,e user_ta.c and ldelf_loader.c, So add an argument to thread_enter_user_mode which is uctx

Please suggest what would be better thing to do.

Reply:
The approach here looks good to me.